packages/plugins/other/visitor-plugin-common/src/client-side-base-visitor.ts

```diff
@@ -109,7 +157,7 @@ export class ClientSideBaseVisitor<TRawConfig extends RawClientSideBasePluginCon
 
   protected _includeFragments(fragments: string[]): string {
     if (fragments && fragments.length > 0) {
-      if (this.config.noGraphQLTag) {
+      if (this._getDocumentMode() === 'documentNode') {
```

Comment:
Maybe add all available options of `documentMode` as TypeScript enum?

Reply:
I was using union type - I have changed this to enum in https://github.com/dotansimha/graphql-code-generator/pull/2261/commits/85ed7654b6d49c85053c09642ecbf7ebb3eae3f0